OmiseSDK/FPXFormViewController.swift

```diff
@@ -0,0 +1,217 @@
+        let valid = formFields.reduce(into: true, { (valid, field) in
+            valid = valid && field.isValid
+        })
+
+        return valid || isEmailInputEmpty
+    }
+
+    private var isEmailInputEmpty: Bool {
+        return emailTextField.text?.trimmingCharacters(in: CharacterSet.whitespaces) == ""
+    }
+
+    @IBInspectable @objc var preferredPrimaryColor: UIColor? {
+        didSet {
+            applyPrimaryColor()
+        }
+    }
+
+    @IBInspectable @objc var preferredSecondaryColor: UIColor? {
+        didSet {
+            applySecondaryColor()
+        }
+    }
+
+    var currentEditingTextField: OmiseTextField?
+
+    @IBOutlet var contentView: UIScrollView!
+
+    @IBOutlet var emailTextField: OmiseTextField!
+    @IBOutlet var submitButton: MainActionButton!
+    @IBOutlet var requestingIndicatorView: UIActivityIndicatorView!
+
+    @IBOutlet var errorLabel: UILabel!
+
+    @IBOutlet var formLabels: [UILabel]!
+    @IBOutlet var formFields: [OmiseTextField]!
+
+    @IBOutlet var formFieldsAccessoryView: UIToolbar!
+    @IBOutlet var gotoPreviousFieldBarButtonItem: UIBarButtonItem!
+    @IBOutlet var gotoNextFieldBarButtonItem: UIBarButtonItem!
+    @IBOutlet var doneEditingBarButtonItem: UIBarButtonItem!
+
+    override func loadView() {
+        super.loadView()
+
+        view.backgroundColor = .background
+        formFieldsAccessoryView.barTintColor = .formAccessoryBarTintColor
+
+        submitButton.defaultBackgroundColor = .omise
+        submitButton.disabledBackgroundColor = .line
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        applyPrimaryColor()
+        applySecondaryColor()
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+
+        formFields.forEach({
+            $0.inputAccessoryView = formFieldsAccessoryView
+        })
+
+        formFields.forEach({
+            $0.adjustsFontForContentSizeCategory = true
+        })
+        formLabels.forEach({
+            $0.adjustsFontForContentSizeCategory = true
+        })
+        submitButton.titleLabel?.adjustsFontForContentSizeCategory = true
+
+        if  #available(iOS 11, *) {
+            // We'll leave the adjusting scroll view insets job for iOS 11 and later to the layoutMargins + safeAreaInsets here
+        } else {
+            automaticallyAdjustsScrollViewInsets = true
+        }
+
+        NotificationCenter.default.addObserver(
+            self, selector:#selector(keyboardWillChangeFrame(_:)),
+            name: NotificationKeyboardWillChangeFrameNotification, object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self, selector:#selector(keyboardWillHide(_:)),
+            name: NotificationKeyboardWillHideFrameNotification, object: nil
+        )
+
+        emailTextField.validator = try! NSRegularExpression(pattern: "\\A[\\w.+-]+@[a-z\\d.-]+\\.[a-z]{2,}\\z", options: [.caseInsensitive])
```

Comment:
Please also avoid using `!` here, `?` will work fine:
```suggestion
        emailTextField.validator = try? NSRegularExpression(pattern: "\\A[\\w.+-]+@[a-z\\d.-]+\\.[a-z]{2,}\\z", options: [.caseInsensitive])
```